lib/telemetry/tracer.ts

```
@@ -0,0 +1,111 @@
+  serviceName: string;
+  attributes: string;
+}) => {
+  const { ZoneContextManager } = await import('@opentelemetry/context-zone');
+
+  const defaultResource = resourceFromAttributes({
+    [ATTR_SERVICE_NAME]: serviceName,
+    ...(attributes ?? '').split(',').reduce(
+      (prev: Record<string, string>, curr: string) => {
+        const [key, value] = curr.split('=');
+        return {
+          ...prev,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)